engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/CollaborationDeploymentTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.deployment.model.validation;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.engine.util.client.DeploymentClient;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.DeploymentIntent;
+import io.camunda.zeebe.protocol.record.value.DeploymentRecordValue;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.Function;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.mockito.verification.VerificationWithTimeout;
+
+@RunWith(Parameterized.class)
+public final class CollaborationDeploymentTest {
+
+  private static final VerificationWithTimeout VERIFICATION_TIMEOUT =
+      timeout(Duration.ofSeconds(1).toMillis());
+
+  @Rule public final EngineRule engine = EngineRule.singlePartition();
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @Parameter(0)
+  public String description;
+
+  @Parameter(1)
+  public Function<DeploymentClient, Record<DeploymentRecordValue>> performDeployment;
+
+  @Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(
+        new Object[] {"collaboration with NPE error one", deploy("/processes/issue-9400-NPE.bpmn")},
+        new Object[] {
+          "collaboration with NPE error two",
+          deploy("/processes/issue-9400-carPolicyRequestProcess.bpmn")
+        },
```

Comment:
🔧 I think we could remove this specific scenario. The other 2 processes already reproduce the problem. There is no need for the complex process. If the test fails it only provides confusing when debugging.

Reply:
Done.